core/src/main/java/org/bitcoinj/base/AddressParser.java

```diff
@@ -39,6 +39,44 @@ public interface AddressParser {
      */
     Address parseAddress(String addressString, Network network) throws AddressFormatException;
 
+    /**
+     * Parse an address for any known/configured network
+     * @param addressString string representation of address
+     * @return A validated address object
+     * @throws AddressFormatException invalid address string
+     * @deprecated Use {@link #parseAddress(String)}
+     */
+    @Deprecated
+    default Address parseAddressAnyNetwork(String addressString) throws AddressFormatException {
+        return parseAddress(addressString);
+    }
+
+    /**
+     * Functional interface for address parsing. It takes a single parameter, like {@link AddressParser#parseAddress(String)}
+     * but its behavior is context-specific. This interface may be
+     * implemented by creating a partial application of ({@link AddressParser#parseAddress(String, Network)} providing
+     * a fixed value for {@link Network}. Or it may behave more like {@link #parseAddress(String)} with a context-specific
+     * list of networks.
+     */
+    @FunctionalInterface
+    interface Simple {
+        /**
+         * Parse an address in a context-specific way (i.e. with a configured list of valid networks)
+         * @param addressString string representation of address
+         * @return A validated address object
+         * @throws AddressFormatException invalid address string or not valid for network (provided by context)
+         */
+        Address parseAddress(String addressString) throws AddressFormatException;
+    }
+
+    /**
+     * Functional interface for strict parsing. It takes a single parameter, like {@link AddressParser#parseAddress(String)}
+     * but is used in a context where a specific {@link Network} has been specified. This interface may be
+     * implemented by creating a partial application of ({@link AddressParser#parseAddress(String, Network)} providing
+     * a fixed value for {@link Network}.
+     * @deprecated You probably want to use {@link AddressParser.Simple}, though the semantics of <q>strict</q> no longer apply.
+     */
+    @Deprecated
```

Comment:
Again, do we need the deprecation?

Reply:
See above.

Reply:
I added a comment on the `@Deprecation` line noting that the deprecation should be removed _before_ 0.17 final.